mkdocs/nav.py

```diff
@@ -186,6 +187,18 @@ def set_canonical_url(self, base):
             base += '/'
         self.canonical_url = utils.urljoin(base, self.abs_url.lstrip('/'))
 
+    def set_edit_url(self, repo_url, edit_uri):
+        if not repo_url.endswith('/'):
+            repo_url += '/'
+        if not edit_uri:
+            self.edit_url = repo_url
+        else:
+            if not edit_uri.endswith('/'):
+                edit_uri += '/'
+            self.edit_url = utils.urljoin(
+                repo_url+edit_uri,
+                self.input_path.lstrip('/'))
+
```

Comment:
1. I was cribbing from `set_canonical_url()`. I can remove those bits if that is not necessary. It gets a little tricky because `utils.urljoin()` only joins a pair of strings, not a list. i.e. this no worky:
```
self.edit_url = utils.urljoin(repo_url, edit_uri, self.input_path.lstrip('/'))
```
So I have to call `urljoin` twice, or make sure that `repo_url` and `edit_url` are properly constructed so I can concatenate them.

2. Line 200 adds `self.input_path`... I am actually building my docs locally to prove to myself that the patch works...